src/DocumentSession.h

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2021, Kung-Yao Hsieh, Shang-Chen LIN, Yi-Hang Xie. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef HTTPLIB_SRC_DOCUMENTSESSION_H
+#define HTTPLIB_SRC_DOCUMENTSESSION_H
+
+#include <unordered_set>
+#include <string>
+#include <memory>
+#include <mutex>
+
+class DocumentManager;
+class SocketSession;
+
+/** @brief Manage the operation between SocketSession and DocumentManager. */
+class DocumentSession {
+private:
+    /** @brief Store DocumentManager pointer. It is used to synchronize datas. */
+    DocumentManager* doc_manager_;
+    /** @brief Store SocketSession object pointer. */
+    std::unordered_set<std::shared_ptr<SocketSession>> sessions_;
```

Comment:
Why do you need hash set? If it's really needed, please add hash for it.

Reply:
We don't need hash. Instead, we use set to replace unordered_set.